### ISSUE TITLE
`RegisterContentPresenter` should always return true

### ISSUE THREAD

**Issue by amwx** [AvaloniaUI/Avalonia#9811]
**Is your feature request related to a problem? Please describe.**
I noticed an issue where a custom `ContentControl` (specifically my `Frame` control) was significantly slowing down runtime swapping of resources (like 300-350% slower) because `Frame` isn't currently registering it's ContentPresenter. This is in part because I don't use the `PART_` naming prefix for template items, which is the only condition the base RegisterContentPresenter method will return true:

https://github.com/AvaloniaUI/Avalonia/blob/6cfa75aaf3cf8c2be841fa9433f5ff71d33e4f7c/src/Avalonia.Controls/ContentControl.cs#L108-L117

I still don't understand why this was the issue, as the logical tree (from devtools) still looked like it was set up correctly & styling applied normally, but it leads me to this:

From what I can tell, this method only serves the purpose of registering the parent ContentControl to the ContentPresenter. (From the code above, it also sets the presenter property on ContentControl, but only if you use that specific name). So is this method really even needed? Is there any case where you'd put a ContentPresenter in a ContentControl and wouldn't want it registered to the parent ContentControl? 

There's also the issue of only registering if you name it `PART_ContentPresenter`, which is dangerous. Regardless of the `PART_` prefix preference, it's possible the name may be desired to be something different to be more specific. Or there could be multiple ContentPresenter for multiple "content-like" properties. 

So IMO, the base method in ContentControl should always return true to ensure a ContentPresenter is registered properly. And, the `Presenter` property on ContentControl should be `protected set` so derived controls can set this accordingly.

**Comment by kekekeks** [comment]
- ContentControl only supports one presenter and there are places in the code base that assume that it can only have one presenter (like automation support)
- ContentControl, like every other templated control, expects its single expected template child to have a certain name


So it seems that `ContentControl` is just not a suitable base class for your needs. You can have your own `IContentPresenterHost`  implementation that would always return `true`.

**Comment by kekekeks** [comment]
We should probably add a `protected` setter to `Presenter` property though, since otherwise the `virtual` flag on the `RegisterContentPresenter` doesn't really make sense.

**Comment by amwx** [comment]
> ContentControl only supports one presenter and there are places in the code base that assume that it can only have one presenter (like automation support)

Yes, ContentControl can only have one presenter for its main content, but certain ContentControl derived controls can have more than one "content-like" property and therefore more than one ContentPresenter in its template. An example is SplitView, which registers the main ContentPresenter, and the one used in the pane:
https://github.com/AvaloniaUI/Avalonia/blob/d7414d39d83e213e3845853877e7ac9ea90b5c8b/src/Avalonia.Controls/SplitView.cs#L318-L328

Obviously the `Presenter` property can only be set to one, but based on my understanding of what `RegisterContentPresenter` does, that's not the main job and setting that property here is more convenience - it's more on hooking the ContentPresenter up to the ContentControl for something that I can't figure out (logical tree related?) and so always returning true here I don't think has a downside - and there's obviously has an impact if you don't register the presenter.


> ContentControl, like every other templated control, expects its single expected template child to have a certain name

IMO, ContentControl should be treated more like a primitive control (not in the namespace sense) and therefore should NOT enforce template naming at this level. It's more a building block for derived controls like Button or List items and generally not used on its own. Derived controls should be where the template naming is enforced - as there may be more specific names desired at that level rather than something generic like "[PART_]ContentPresenter"

**Comment by kekekeks** [comment]
>An example is SplitView

Looking at the code, it seems that automation support for `SplitView` might be broken, since it doesn't have its own automation peer and `PART_PanePresenter` isn't handled by `ContentControlAutomationPeer`
